packages/react/package.json

```diff
@@ -72,7 +72,8 @@
     "qrcode": "1.5.0",
     "react-generate-context": "1.0.1",
     "react-map-gl": "7.0.15",
-    "tslib": "2.4.0"
+    "tslib": "2.4.0",
+    "xstate": "^4.33.6"
```

Comment:
`xstate` becomes a core dependency of `@aws-amplify/ui-react` and will be used by `@aws-amplify/ui`, `@aws-amplify/ui-react-core`, and `@xstate/react`.